src/snowcli/cli/nativeapp/init.py

```diff
@@ -220,9 +188,13 @@ def _init_with_url_and_no_template(
         # Remove all git history
         rmtree(target_directory.joinpath(".git").resolve())
 
-        # Non-Snowflake git URLs should not have jinja files in their directory structure.
-        # If they do, snowCLI is not responsible for rendering them during init.
-        # If the SNOWFLAKELABS_GITHUB_URL is provided here, rendering is additionally skipped as there is no jinja file at the root.
+        # While non-Snowflake git URLs may have jinja files in their directory structure,
+        # snowCLI is not responsible for rendering them during init because they might have variables other than {{project_name}}.
```

Comment:
One thing that can be done is passing `undefined=jinja2.StrictUndefined` which will fail early in the case where there are undefined variables. Might be a path to having other git repos use jinja2?